arthur/tasks.py

```diff
@@ -177,14 +180,14 @@ def add(self, task_id, backend, category, backend_args,
         """
         self._rwlock.writer_acquire()
 
-        if task_id in self._tasks:
+        if task_id in [k.decode("utf-8") for k in self.conn.keys()]:
             self._rwlock.writer_release()
             raise AlreadyExistsError(element=str(task_id))
 
         task = Task(task_id, backend, category, backend_args,
                     archiving_cfg=archiving_cfg,
                     scheduling_cfg=scheduling_cfg)
-        self._tasks[task_id] = task
+        self.conn.set(task_id, pickle.dumps(task))
```

Comment:
I'm not a big fun of this solution but it might work for now.

Reply:
In the case `self.conn.set` fails due to any kind of error (for example, connection errors), this will lock the `rwlock` object forever because it is never released. We should work with `try...except..finally` blocks to fix this problem.
This wasn't a problem before because tasks were stored in memory.